packages/shopping/src/services/validator.ts

```diff
@@ -20,3 +22,26 @@ export function validateCredentials(credentials: Credentials) {
     );
   }
 }
+
+export function validateKeyPassword(keyAndPassword: KeyAndPassword) {
+  // Validate Password Length
+  if (!keyAndPassword.password || keyAndPassword.password.length < 8) {
+    throw new HttpErrors.UnprocessableEntity(
+      'password must be minimum 8 characters',
+    );
+  }
+
+  if (keyAndPassword.password !== keyAndPassword.confirmPassword) {
+    throw new HttpErrors.UnprocessableEntity(
+      'password and confirmation password do not match',
+    );
+  }
+
+  if (
+    _.isEmpty(keyAndPassword.resetKey) ||
+    keyAndPassword.resetKey.length === 0 ||
+    keyAndPassword.resetKey.trim() === ''
```

Comment:
Let's keep code consistent with L28 and lose the unnecessary lodash.

```suggestion
    !keyAndPassword.resetKey ||
    keyAndPassword.resetKey.trim().length === 0
```